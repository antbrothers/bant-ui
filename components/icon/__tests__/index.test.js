/*
 * @Author: linjianxi
 * @Date: 2020-01-09 16:16:34
 * @LastEditTime : 2020-01-09 16:38:53
 * @Description: file content
 */
import React from 'react'
import { render, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Icon from '../index'

describe("<Icon/>", () => {
    it("should render a <Icon/> components", () => {
        const wrapper = render(
            <div>
                <Icon type="loading" size="sm"></Icon>
                <Icon type="check" color="D9AF5C" size="sm"></Icon>
                <Icon type="check-circle-o" size="sm"></Icon>
                <Icon type="check-circle" color="green" size="sm"></Icon>
                <Icon type="cross" size="sm"></Icon>
                <Icon type="cross-circle" size="sm"></Icon>
                <Icon type="cross-circle-o" color="red" size="sm"></Icon>
                <Icon type="up" size="sm"></Icon>
                <Icon type="down" size="sm"></Icon>
                <Icon type="left" size="sm"></Icon>
                <Icon type="right" size="sm"></Icon>
                <Icon type="ellipsis" size="sm"></Icon>
                <Icon type="loading" size="xs"></Icon>      
                <Icon type="loading" size="xxs"></Icon>     
                <Icon type="loading" size="sm"></Icon>       
                <Icon type="loading" size="md"></Icon>     
                <Icon type="loading" size="lg"></Icon>   

            </div>
        )
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})