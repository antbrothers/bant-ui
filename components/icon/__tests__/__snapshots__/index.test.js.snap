// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> should render a <Icon/> components 1`] = `
<div>
  <svg
    class="bant-icon bant-icon-loading bant-icon-sm"
  >
    <use
      href="#loading"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-check bant-icon-sm"
    style="color:D9AF5C"
  >
    <use
      href="#check"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-check-circle-o bant-icon-sm"
  >
    <use
      href="#check-circle-o"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-check-circle bant-icon-sm"
    style="color:green"
  >
    <use
      href="#check-circle"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-cross bant-icon-sm"
  >
    <use
      href="#cross"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-cross-circle bant-icon-sm"
  >
    <use
      href="#cross-circle"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-cross-circle-o bant-icon-sm"
    style="color:red"
  >
    <use
      href="#cross-circle-o"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-up bant-icon-sm"
  >
    <use
      href="#up"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-down bant-icon-sm"
  >
    <use
      href="#down"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-left bant-icon-sm"
  >
    <use
      href="#left"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-right bant-icon-sm"
  >
    <use
      href="#right"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-ellipsis bant-icon-sm"
  >
    <use
      href="#ellipsis"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-loading bant-icon-xs"
  >
    <use
      href="#loading"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-loading bant-icon-xxs"
  >
    <use
      href="#loading"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-loading bant-icon-sm"
  >
    <use
      href="#loading"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-loading bant-icon-md"
  >
    <use
      href="#loading"
    />
  </svg>
  <svg
    class="bant-icon bant-icon-loading bant-icon-lg"
  >
    <use
      href="#loading"
    />
  </svg>
</div>
`;
